modules/client/src/internal/client/methods.ts

```diff
@@ -410,17 +410,41 @@ export class ClientMethods extends ClientCore implements IClientMethods {
 
     // filter specified installation
     const { pluginInstallationIndex = 0 } = params;
-    const filteredInstallation = iplugin.installations.filter(
-      (installation) => {
-        return installation.appliedVersion.build ===
-            params.versionTag.build &&
-          installation.appliedVersion.release.release ===
-            params.versionTag.release;
-      },
-    )[pluginInstallationIndex];
-    if (!filteredInstallation) {
+    const selectedInstallation = iplugin.installations[pluginInstallationIndex];
+    if (!selectedInstallation) {
       throw new InstallationNotFoundError();
     }
+    // check if metadata is defined
+    if (!selectedInstallation.appliedVersion.metadata) {
+      throw new MissingMetadataError();
```

Comment:
In case a plugin has no metadata, I think the SDK should allow for a way to manually specify it

Reply:
I think this wont be the case, because all plugins deployed with our deploy script have metadata, however one solution would be allowing to pass the pluginUninstallationAbi as an optional parameter the other one is abstracting the encoding functionality from the `prepareUninstallation` function and receive a data parameter

Reply:
Our plugins will be fine, but nothing prevents anyone else from publishing a metadata-less plugin. In such a case, a DAO could not uninstall such plugin through the SDK

Adding an optional parameter sounds good. Alternatively, having a specific function receiving the ABI instead of fetching it.

Reply:
If the end user facing parameters don't look strange, I would add an optional parameter to receive the ABI in case it was not available (for some reason).